sound/soc/sof/ipc.c

```diff
@@ -286,7 +286,7 @@ static int sof_ipc_tx_message_unlocked(struct snd_sof_ipc *ipc, u32 header,
 
 	spin_unlock_irq(&sdev->ipc_lock);
 
-	if (ret < 0) {
+	if (ret) {
```

Comment:
FWIW personally I'm in favour of explicit `ret < 0` tests - that way you don't pick up positive `ret` values "by accident." But I realise, that we have a mix and that we probably should unify towards the more common one...

Reply:
Is this change OK or would you prefer a change like this:
```
	ret = snd_sof_dsp_send_msg(sdev, msg);
	/* Next reply that we receive will be related to this message */
-	if (ret)
+	if (ret >= 0)
		msg->ipc_complete = false;

	spin_unlock_irq(&sdev->ipc_lock);

	if (ret < 0) {
		dev_err_ratelimited(sdev->dev,
				    "error: ipc tx failed with error %d\n",
				    ret);
		return ret;
	}
```

Reply:
this looks like a separate fix we need to review @ujfalusi, let's go with the PR as is for now.